sky/serve/replica_managers.py

```diff
@@ -1112,6 +1112,37 @@ def update_version(self, version: int,
         self.latest_version = version
         self._task_yaml_path = task_yaml_path
 
+        # Update the version for all replicas that have the same config as
+        # the new version (except for the `service` field), as we do not need
+        # to restart those replicas. This can significantly improve the speed
+        # for updating an existing service with only config changes to the
+        # service specs, e.g. scale down the service.
+        # TODO(MaoZiming): Check whether user updates any file in file mounts.
+        new_config = common_utils.read_yaml(os.path.expanduser(task_yaml_path))
+        for key in ['service']:
+            new_config.pop(key)
+        replica_infos = serve_state.get_replica_infos(self._service_name)
```

Comment:
Based on our discussion offline, should we skip the following for loop `if new_config.get('file_mounts', None) is not None or new_config.get('workdir', None) is not None`